prober/utils.go

```diff
@@ -25,32 +25,37 @@ func chooseProtocol(preferredIPProtocol string, target string, registry *prometh
 	registry.MustRegister(probeIPProtocolGauge)
 	registry.MustRegister(probeDNSLookupTimeSeconds)
 
-	if preferredIPProtocol == "ip6" || preferredIPProtocol == "" {
-		preferredIPProtocol = "ip6"
+	if IPProtocol == "ip6" || IPProtocol == "" {
+		IPProtocol = "ip6"
 		fallbackProtocol = "ip4"
 	} else {
-		preferredIPProtocol = "ip4"
+		IPProtocol = "ip4"
 		fallbackProtocol = "ip6"
 	}
 
-	if preferredIPProtocol == "ip6" {
+	if IPProtocol == "ip6" {
 		fallbackProtocol = "ip4"
 	} else {
 		fallbackProtocol = "ip6"
 	}
 
-	level.Info(logger).Log("msg", "Resolving target address", "preferred_ip_protocol", preferredIPProtocol)
+	level.Info(logger).Log("msg", "Resolving target address", "ip_protocol", IPProtocol)
 	resolveStart := time.Now()
 
 	defer func() {
 		lookupTime = time.Since(resolveStart).Seconds()
 		probeDNSLookupTimeSeconds.Add(lookupTime)
 	}()
 
-	ip, err = net.ResolveIPAddr(preferredIPProtocol, target)
+	ip, err = net.ResolveIPAddr(IPProtocol, target)
 	if err != nil {
-		level.Warn(logger).Log("msg", "Resolution with preferred IP protocol failed, attempting fallback protocol", "fallback_protocol", fallbackProtocol, "err", err)
-		ip, err = net.ResolveIPAddr(fallbackProtocol, target)
+		if fallbackIPProtocol == false {
+			level.Error(logger).Log("msg", "Resolution with IP protocol failed (fallback IP protocol is FALSE): err", err)
```

Comment:
fallback_ip_protocol would be easier for users to find, and the FALSE could be false.